src/stream_wrap.cc

```diff
@@ -276,14 +276,15 @@ void StreamWrap::SetBlocking(const FunctionCallbackInfo<Value>& args) {
 
 int StreamWrap::DoShutdown(ShutdownWrap* req_wrap) {
   int err;
-  err = uv_shutdown(&req_wrap->req_, stream(), AfterShutdown);
+  err = uv_shutdown(req_wrap->req(), stream(), AfterShutdown);
   req_wrap->Dispatched();
   return err;
 }
 
 
 void StreamWrap::AfterShutdown(uv_shutdown_t* req, int status) {
-  ShutdownWrap* req_wrap = ContainerOf(&ShutdownWrap::req_, req);
+  ShutdownWrap* req_wrap = static_cast<ShutdownWrap*>(req->data);
+  CHECK_NE(req_wrap, nullptr);
```

Comment:
I've added a commit with `from_req()` methods. Let me know if this was what you had in mind. Thanks